AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSPluginsCore
+@testable import AWSDataStoreCategoryPlugin
+
+class SyncEventEmitterTests: XCTestCase {
+    var initialSyncOrchestrator: MockAWSInitialSyncOrchestrator?
+    var reconciliationQueue: MockAWSIncomingEventReconciliationQueue?
+    var syncEventEmitter: SyncEventEmitter?
+    var expectedModelSyncedEventPayloads: [ModelSyncedEvent]!
+
+    override func setUp() {
+        super.setUp()
+
+        ModelRegistry.reset()
+        MockModelReconciliationQueue.reset()
+        MockAWSInitialSyncOrchestrator.reset()
+
+        expectedModelSyncedEventPayloads = [ModelSyncedEvent(modelName: "Post",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0),
+                                            ModelSyncedEvent(modelName: "Comment",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0)]
+    }
+
+    /// - Given: A SyncEventEmitter
+    /// - When:
+    ///    - One model is registered
+    ///    - Perform an initial sync
+    /// - Then:
+    ///    - One modelSynced event should be received
+    ///    - One syncQueriesReady event should be received
+    func testModelSyncedAndSyncQueriesReadyWithOneModelRegistered() throws {
+        let modelSyncedReceived = expectation(description: "modelSynced received")
+        let syncQueriesReadyReceived = expectation(description: "syncQueriesReady received")
+
+        ModelRegistry.register(modelType: Post.self)
```

Comment:
can this be done in the setup?

Reply:
For the 1st test, I want `Post` to be registered, for the 2nd test, I want `Post` and `Comment` be registered.

So that I decided to separate them. If `Comment` is also registered for the 1st test, it's going to fail the test